src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/IRazorProjectService.cs

```diff
@@ -14,7 +14,7 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
 
 internal interface IRazorProjectService
 {
-    Task AddDocumentAsync(string filePath, CancellationToken cancellationToken);
+    Task AddDocumentToMiscProjectAsync(string filePath, CancellationToken cancellationToken);
```

Comment:
Somewhere in this PR, I'd love to see comments to explain why this is the way it is. Your PR description was good, and I think it'd be good to capture it in the code.

Reply:
>  it makes me a little sad that the symmetry is lost

RemoveDocument is in a similar position as AddDocument was: It's only called by the file watcher, and in lots and lots of tests. Given that it's removing documents though, its not having to guess so its a little more reasonable, though I'd still be a fan of having the client be the source of truth (via project.razor.vs.bin), I didn't want to break into that jail right now. Plus it would mean RemoveDocument would become MoveDocumentsToMiscProject so we'd lose symmetry anyway

> Somewhere in this PR, I'd love to see comments to explain why this is the way it is. Your PR description was good, and I think it'd be good to capture it in the code.

That's why I renamed the method, but I'll add some comments around explaining why we don't want to try to do anything smarter, lest someone is tempted to repeat history :)